client/src/components/common/styledButton/StyledButton.js

```diff
@@ -4,7 +4,7 @@ import { Button } from 'semantic-ui-react';
 import PropTypes from 'prop-types';
 
 const StyledButton = ({
-  text, url, size, color, link, inverted, fluid,
+  text, url, size, color, link, inverted, fluid, onClick,
```

Comment:
```suggestion
  const StyledButton = ({ url, link , ...props }) => {
    const buttonProps = { to: url, as: link ? Link: Button }
    return <Button {...buttonProps} {...props}>{text}</Button>
  };
```
By above approach you have to set less defaultProps and can use more props of the button